pkg/cluster/api/pdapi.go

```diff
@@ -535,6 +536,110 @@ func (pc *PDClient) EvictStoreLeader(host string, retryOpt *utils.RetryOption, c
 	return nil
 }
 
+// RecoverStoreLeader waits for some leaders to transfer back.
+//
+// Currently, recoverStoreLeader will be considered as succeed in any of the following case
+//
+//  1. 2/3 of leaders are already transferred back.
+//
+//  2. Original leader count is less than 57.
+//
+//     Since PD considers it as balance when the leader count delta is less than 10, so
+//     these two conditions should be taken into consideration
+//
+//     - When the original leader count is less than 20, there is possibility that
+//     no leader will transfer back.
+//     For example: The target store's leader count is 19. Other stores' leader count are 9.
+//     There are 20 stores in total. In this case, there may be no leader to transfer back.
+//
+//     - When the leader count is less than 57, there is possibility that only less than 2/3
+//     leaders are transfered back. `(N-10-9 >= 2/3*N) -> (N>=57)`.
```

Comment:
```suggestion
//     leaders are transfered back. `(N-10-9 > 2/3*N) -> (N>57)`.
```

Reply:
maybe we need to verify it by pd-simulator

Reply:
I think N can be equal to 57. When it is 57, it should has at least 38 leader after balance and 38/57==2/3. :thinking: 

I'll take a look about the verification.

Reply:
I think we just need to know the **accurate** value in our mind, and we can set the value to a larger number in tiup code. The tiup code does not need to be strongly coupled with the PD code. 

Moreover, clusters which have small number of leaders is supposed to has low pressure, and this enhancement may be unnecessary for them. Clusters in production env usually has thousands of leaders. 

So, I think we can adjust the code to the following. What do you think? @lhy1024 @AstroProfundis 
```golang
if originalCount < 200 {
    return nil
}
```

---------

In the future, tidb may support big sized regions. At that time, I think PD should provider some API to help other tools (like tiup/tidb-operators) integrate easily.

Reply:
I think 200 is better than 57.

Reply:
I am not sure whether `leaderTolerantSizeRatio`  will be changed by dynamic region. cc @bufferflies

Reply:
Can we merge this PR first? /cc @lhy1024 @AstroProfundis 

IMO, even `leaderTolerantSizeRatio` is changed, the current threshold (200) is suitable.